### PR TITLE
osdc/Objecter: unify disparate EAGAIN handling paths into one

### PR DESCRIPTION
**Pull request by liewegas**
We had two different code paths for EAGAIN that both
seemed to resend the same request again (with a new
tid), but implemneted it in two different ways. Further,
the first one was conditioned on the BALANCE_READS
flag but not LOCALIZED_READS, which should (I think) be
equivalent in this context.

Unify them into a single path.

Signed-off-by: Sage Weil <sage@redhat.com>

### CODE REVIEW

File: src/osdc/Objecter.cc [hunk 1]
```diff
@@ -3415,30 +3415,16 @@ void Objecter::handle_osd_op_reply(MOSDOpReply *m)
...
-
-    _send_op(op);
+    if (op->onfinish)
+      num_in_flight--;
+    _session_op_remove(s, op);
     sl.unlock();
     put_session(s);
+
+    op->tid = 0;
+    op->target.flags &= ~CEPH_OSD_FLAG_BALANCE_READS;
```
Comment (dillaman) [inline src/osdc/Objecter.cc]: Nit: remove localize read flag as well?

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2017-07-29_05:31:06-rados-wip-kefu-testing-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2017-07-30_00:01:51-rados-wip-kefu-testing-distro-basic-smithi/

**Comment by tchaikov** [comment]
@dillaman sorry, i jumped the gun. did you get a chance to run this patch thru RBD fsx tests? or do we have a tracker ticket tracking that issue?

**Comment by dillaman** [comment]
@tchaikov No worries -- I'll check the test runs on Monday

**Pull request merged**